// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package scql.pb;

option go_package = "proto-gen/scql";

enum Code {
  // Not an error; returned on success
  OK = 0;

  // =====================
  // 100-199 Request Error
  // =====================

  // General bad request
  BAD_REQUEST = 100;
  // Failed to authenticate user credential
  UNAUTHENTICATED = 101;
  // Parsing SQL error, returned on invalid SQL
  SQL_PARSE_ERROR = 102;
  INVALID_ARGUMENT = 103;
  // Query result is not ready, please try again later
  NOT_READY = 104;

  // The user does not have permission to execute the DDL.
  // For example, the user try to drop other's table.
  DDL_PERMISSION_DENIED = 131;

  // General not found error
  NOT_FOUND = 140;
  // SCDB session not found
  SESSION_NOT_FOUND = 141;

  // Query CCL check failed
  CCL_CHECK_FAILED = 160;

  // The P2P internal exchange error
  // Local storage info is not equal to storage info in other parties
  DATA_INCONSISTENCY = 170;

  // =========================
  // 200-299 Dependence Error
  // =========================

  // SCDB DB error
  STORAGE_ERROR = 201;

  // ==============================
  // 300-399 Server Internal Error
  // ==============================

  // UNKNOWN Server Internal Error
  INTERNAL = 300;

  // 320-339 executing execution graph error
  UNKNOWN_ENGINE_ERROR = 320;

  ENGINE_RUNSQL_ERROR = 332;

  // Feature not supported
  NOT_SUPPORTED = 340;
};